sponsorship_compassion/models/contracts.py

```diff
@@ -522,10 +522,12 @@ def write(self, vals):
 
         updated_correspondents = self.env[self._name]
         if "correspondent_id" in vals:
-            old_correspondents = self.mapped("correspondent_id")
-            updated_correspondents = self._on_change_correspondant(
-                vals["correspondent_id"]
-            )
+            for record in self:
+                if record.global_id:
+                    old_correspondents = record.correspondent_id
+                    record._remove_correspondent(old_correspondents)
+                    self.correspondent_id.update_number_sponsorships()
```

Comment:
This should be moved inside the `_remove_correspondent` method, just before the commit.